examples/webgl_buffergeometry_instancing_standard.html

```diff
@@ -0,0 +1,531 @@
+				geometry.addAttribute( 'instanceNormalMatrixColumn2', new THREE.InstancedBufferAttribute( new Float32Array( objectNum * 3 ), 3 ) );
+				geometry.addAttribute( 'instanceColor', new THREE.InstancedBufferAttribute( new Float32Array( instanceColors ), 3 ) );
+				geometry.maxInstancedCount = objectNum;
+
+
+				// Create InstancedMesh
+
+				var mesh = new InstancedMesh(
+					geometry,
+					new THREE.MeshStandardMaterial( { metalness: 0.2, color: 0xffffff } )
+				);
+				mesh.castShadow = true;
+				mesh.receiveShadow = true;
+				scene.add( mesh );
+
+
+				// Inject code for Instancing to built-in MeshStandardMaterial shader
+
+				var vertexParsChunk = [
+					'uniform float time;',
+					'attribute vec4 instanceMatrixColumn0;',
+					'attribute vec4 instanceMatrixColumn1;',
+					'attribute vec4 instanceMatrixColumn2;',
+					'attribute vec4 instanceMatrixColumn3;'
+				].join( '\n' ) + '\n';
+
+				var normalParsChunk = [
+					'attribute vec3 instanceNormalMatrixColumn0;',
+					'attribute vec3 instanceNormalMatrixColumn1;',
+					'attribute vec3 instanceNormalMatrixColumn2;'
+				].join( '\n' ) + '\n';
+
+				var colorParsChunk = [
+					'attribute vec3 instanceColor;',
+					'varying vec3 vInstanceColor;'
+				].join( '\n' ) + '\n';
+
+				var vertexChunk = [
+					'mat4 instanceMatrix = mat4(',
+					'	instanceMatrixColumn0,',
+					'	instanceMatrixColumn1,',
+					'	instanceMatrixColumn2,',
+					'	instanceMatrixColumn3',
+					');',
+					'vec3 transformed = ( instanceMatrix * vec4( position, 1.0 ) ).xyz;'
+				].join( '\n' ) + '\n';
+
+				var instanceColorChunk = [
+					'vInstanceColor = instanceColor;'
+				].join( '\n' ) + '\n';
+
+				var normalChunk = [
+					'mat3 instanceNormalMatrix = mat3(',
+					'	instanceNormalMatrixColumn0,',
+					'	instanceNormalMatrixColumn1,',
+					'	instanceNormalMatrixColumn2',
+					');',
+					'objectNormal = instanceNormalMatrix * objectNormal;'
+				].join( '\n' ) + '\n';
+
+				var fragmentParsChunk = [
+					'varying vec3 vInstanceColor;'
+				].join( '\n' ) + '\n';
+
+				var colorChunk = [
+					'vec4 diffuseColor = vec4( diffuse * vInstanceColor, opacity );'
+				].join( '\n' ) + '\n';
+
+				mesh.material.onBeforeCompile = function ( shader ) {
+
+					// console.log( shader.uniforms );
+					// console.log( shader.vertexShader );
+					// console.log( shader.fragmentShader );
+
+					shader.vertexShader = shader.vertexShader
+						.replace( '#include <common>\n', '#include <common>\n' + vertexParsChunk + normalParsChunk + colorParsChunk )
+						.replace( '#include <defaultnormal_vertex>\n', normalChunk + '#include <defaultnormal_vertex>\n' )
+						.replace( '#include <begin_vertex>\n', vertexChunk + instanceColorChunk );
+
+					shader.fragmentShader = shader.fragmentShader
+						.replace( '#include <common>\n', '#include <common>' + fragmentParsChunk )
+						.replace( 'vec4 diffuseColor = vec4( diffuse, opacity );\n', colorChunk )
+
```

Comment:
Note: As you know code injection may be fragile because if we update built-in material it can fail. Node-based system can do more safely.